nifi-toolkit/nifi-toolkit-tls/src/main/java/org/apache/nifi/toolkit/tls/standalone/TlsToolkitStandalone.java

```diff
@@ -177,6 +184,16 @@ public void createNifiKeystoresAndTrustStores(StandaloneConfig standaloneConfig)
             File keystore = new File(hostDir, "keystore." + tlsClientConfig.getKeyStoreType().toLowerCase());
             File truststore = new File(hostDir, "truststore." + tlsClientConfig.getTrustStoreType().toLowerCase());
 
+            // Adjust the SANs when ranges match.
+            if (domainAlternativeNames.size() == 1) {
+                tlsClientConfig.setDomainAlternativeNames(Collections.singletonList(domainAlternativeNames.get(0)));
+            } else if (domainAlternativeNames.size() == instanceDefinitions.size()) {
+                tlsClientConfig.setDomainAlternativeNames(Collections.singletonList(domainAlternativeNames.get(instanceIndex)));
+                logger.info("Using alternate name " + domainAlternativeNames.get(instanceIndex) + " with hostname " + hostname + ".");
+            } else {
+                logger.info("Hostname count does not match given alternate name count.  Verify names in resulting certificate.");
```

Comment:
Might want to escalate this to `logger.warn` as it would be an edge case (not even sure we know how to handle it) when a non-static domain alternative name pattern is applied but the range does not match the hostname pattern count.

Reply:
Will update to `logger.warn`